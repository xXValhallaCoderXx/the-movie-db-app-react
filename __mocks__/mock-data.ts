import {
  ISelectedMovie,
  IMovieDetail,
  ICast,
  IReviews
} from "../src/shared/types";

export const castData: ICast[] = [
  {
    cast_id: 1,
    character: "Pikachu",
    name: "Pikachuuuuu",
    profile_path: "/img.jpg"
  }
];

export const reviewData: IReviews[] = [
  {
    author: "Ash Catchem",
    content: "This is the review",
    url: "www.urlreview.com"
  }
];

export const searchMovieResponse: IMovieDetail[] = [
  {
    id: "1",
    backdrop_path: "/img.jpg",
    budget: 2000,
    homepage: "www.marvel.com",
    overview: "This is the overview",
    imdb_id: "1234",
    poster_path: "/img.jpg",
    revenue: 2000000,
    release_date: "",
    runtime: 143,
    status: "released",
    tagline: "To infinity and beyond",
    title: "Avengers",
    genres: [
      {
        id: 28,
        name: "Action"
      }
    ]
  }
];

export const mockPaginationData = (data): IMovieDetail[] => {
  const mockData: IMovieDetail[] = [];
  for (let i = 0; i < 61; i++) {
    mockData.push(data);
  }
  return mockData;
};

export const invalidSearchResponse: any[] = [
  {
    id: "1",
    backdrop_path: "/img.jpg",
    budget: 2000,
    homepage: "www.marvel.com",
    overview: "This is the overview",
    imdb_id: "1234",
    poster_path: "/img.jpg",
    revenue: 2000000,
    release_date: "",
    runtime: 143,
    status: "released",
    tagline: "To infinity and beyond",
    genres: [
      {
        id: 28,
        name: "Action"
      }
    ]
  }
];

export const searchResults = [
  {
    id: 603,
    title: "The Matrix"
  }
];

export const selectedMovie: ISelectedMovie = {
  cast: [
    {
      cast_id: 34,
      character: "Thomas A. Anderson / Neo",
      name: "Keanu Reeves",
      profile_path: "/bOlYWhVuOiU6azC4Bw6zlXZ5QTC.jpg"
    }
  ],
  info: {
    id: "1",
    backdrop_path: "/img.jpg",
    budget: 2000,
    homepage: "www.marvel.com",
    overview: "This is the overview",
    imdb_id: "1234",
    poster_path: "/img.jpg",
    revenue: 2000000,
    release_date: "",
    runtime: 143,
    status: "released",
    tagline: "To infinity and beyond",
    title: "Avengers",
    genres: [
      {
        id: 28,
        name: "Action"
      }
    ]
  },
  reviews: [
    {
      author: "GeekMasher",
      content:
        "The Martix is a great example of a movie that will live for ever or a very log time. The story and concept are out of this world. Keanu Reeves plays his role with utter brilliance, the cast was very well put together and the graphics are still to this day amazing. All in all one of the best movies of all time.",
      url: "https://www.themoviedb.org/review/51429fb419c29552f00f2f7b"
    }
  ],
  similar: [
    {
      id: 605,
      title: "The Matrix Revolutions",
      poster_path: "/sKogjhfs5q3azmpW7DFKKAeLEG8.jpg"
    }
  ]
};
