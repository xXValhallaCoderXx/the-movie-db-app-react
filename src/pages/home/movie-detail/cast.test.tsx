import React from "react";
import {castData} from "../../../../__mocks__/mock-data";
import {shallow} from "enzyme";
import Cast from "./cast";

describe("Movie Details - Cast", () => {
  it("should render correctly without data", () => {
    const component = shallow(<Cast cast={[]} />);
    expect(component.find("#no-cast").text()).toEqual("No Cast Data");
    expect(component).toMatchSnapshot();
  });
  it("should render correctly with data", () => {
    const component = shallow(<Cast cast={castData} />);
    expect(component.find("#cast-avatar-name").text()).toEqual(
      castData[0].name
    );
    expect(component.find("#cast-avatar-character").text()).toEqual(
      castData[0].character
    );
    expect(component).toMatchSnapshot();
  });
});
