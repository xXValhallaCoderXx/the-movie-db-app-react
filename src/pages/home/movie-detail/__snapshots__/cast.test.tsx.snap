// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details - Cast should render correctly with data 1`] = `
<div
  className="flex flex-wrap justify-center mt-2"
>
  <div
    className="flex justify-content items-center flex-col p-5"
    id="cast-avatar"
    key="0"
  >
    <img
      className="w-24 h-24 rounded-full"
      src="https://image.tmdb.org/t/p/w264_and_h264_bestv2/img.jpg"
    />
    <p
      className="font-bold"
      id="cast-avatar-name"
    >
      Pikachuuuuu
    </p>
    <p
      className="italic"
      id="cast-avatar-character"
    >
      Pikachu
    </p>
  </div>
</div>
`;

exports[`Movie Details - Cast should render correctly without data 1`] = `
<div
  className="text-center font-bold text-xl"
  id="no-cast"
>
  No Cast Data
</div>
`;
